app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -811,7 +811,10 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
     }
 
     private fun updateQrButton(searchFragmentState: SearchFragmentState) {
-        when (searchFragmentState.searchEngineSource.searchEngine == searchFragmentState.defaultEngine) {
+        val searchEngine = searchFragmentState.searchEngineSource.searchEngine
+        when (
+            searchEngine?.isGeneral == true || searchEngine?.type == SearchEngine.Type.CUSTOM
```

Comment:
@DreVla Wouldn't we only need to check `isGeneral` since https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/feature/search/src/main/java/mozilla/components/feature/search/storage/SearchEngineReader.kt#74 will set `isGeneral = true` for all custom search engines?